snapcraft/cli/store.py

```diff
@@ -101,12 +106,21 @@ def push(snap_file, release):
     If --release is used, the channel map will be displayed after the
     operation takes place.
 
+    If --only-metadata is given, only the snap's metadata information will be
+    pushed to the server (not the whole .snap file). Because of this, --release
+    is not a compatible option to use with it.
+
     \b
     Examples:
         snapcraft push my-snap_0.1_amd64.snap
+        snapcraft push my-snap_0.1_amd64.snap --only-metadata
```

Comment:
hey @matthewpaulthomas or @evandandrea  do you think `--only-metadata` is correct here, it feels strange to me even though it is technically correct. Should this be `--metadata` or something else?

Reply:
I don't think `--only-metadata` is self-explanatory. The option should reflect the developer's intention, not what happens internally (the metadata is sent, the snap is not).

I would be more explicit and use `--overwrite-metadata`.

Reply:
Note however that "overwrite metadata" is not what it does, unless you also pass "force metadata", because pushing metadata can result in conflicts if you modified those fields through the web UI.

Also, these names are stablished in the proper [forum post](https://forum.snapcraft.io/t/mechanisms-for-converging-store-and-snap-metadata/2200), should discuss these there?